pkglistgen/tool.py

```diff
@@ -721,15 +724,15 @@ def update_and_solve_target(
         else:
             url = api.makeurl(['source', project])
             packages = ET.parse(http_GET(url)).getroot()
-            if packages.find(f'entry[@name="{product_package}"]') is None:
+            if packages.find(f'entry[@name="{product_package}"]') is None and not only_update_weakremovers:
                 if not self.dry_run:
                     undelete_package(api.apiurl, project, product_package, 'revive')
                 # TODO disable build.
                 logging.info(f'{product_package} undeleted from {project}, skip dvd until next cycle')
                 return
 
             drop_list = api.item_exists(project, oldrepos)
-            if drop_list and not only_release_packages:
+            if (drop_list and not only_release_packages) or (drop_list and only_update_weakremovers):
```

Comment:
I think this change is unnecessary - the first part is already true, isn't it?

Reply:
ah, good catch!